lib/Weight/MockWeight.hpp

```diff
@@ -0,0 +1,31 @@
+#include "IWeight.hpp"
+
+/**
+ * @brief Mock weight class that inherits from the IWeight interface class.
+ * Used for unit tests so that actual hardware isn't require to test the ZotBins
+ * system.
+ */
+
+class MockWeight : IWeight
+{
+public:
+    /**
+     * @brief Constructs an instance of the MockWeight class.
+     * @param mWeight Represents the current weight of the mock sensor.
+     *  
+     */
+    MockWeight(int32_t mWeight);
```

Comment:
Change `mWeight` to `weight` because the `mWeight` in the parameters is not a member variable